### ISSUE TITLE
Can you share Blish settings across multiple GW2Launcher Profiles

### ISSUE THREAD

**Issue by Kawsmecal** [Healix/Gw2Launcher#335]
Is there a way to do this? I'd like to set up my profiles so stuff like the to-do module is shared on just one Blish opening while allowing paths and stuff to just be set to the active account. I right now have 4 profiles for 2 accounts in GW2Launcher for when I want to swap between which one is on the main monitor and which is on my secondary monitor. It's not really a big deal as I can just not use the module and have something standalone like gw2eff tracking that stuff but I've enjoyed multiboxing to do simple things like farming my home instance it makes it a lot less boring and more engaging to swap between the two between animations of gathering mats. I also want to be able to put both onto the secondary to use my main to help me organize when playing inventory wars with dedupe-o-tron (Please add QoL masteries or just freely given for inventory management Anet I beg you)

**Comment by Healix** [comment]
Blish by default shares its settings between all instances. If you want to give an account different settings, you can specify where the settings folder is located using the -s option.

By default, the option would be:

-s "%userprofile%\Documents\Guild Wars 2\addons\blishhud"

If you were to change 1 account to use:

-s 1

... it would create a "1" folder in the same folder as Blish HUD.exe and only that 1 account would have different settings. If you then added the same option to another account, only those 2 accounts would share the same settings.

If you want to run multiple instance of Blish with different settings, you'll need to specify the mumble link name, which causes Blish to run multiple instances instead of 1 instance that moves to the currently active window. You can set the mumble link name under the launch options and you'd pass that to Blish (-m MumbleLink_%accountid%).

**Comment by Kawsmecal** [comment]
I want the Blish HUD to appear on both clients but have the same shared settings for some that are clones of the same ones (the profiles all with mumble link.

For instance right now I have Main Account (Primary), Alt Account (Secondary), Main Account (Secondary), and Alt Account (Primary).

It'd be nice if there was some way so that the placement of elements line up with either account so HUD elements are in the right places when either Main or Alt are on my primary monitor, while keeping a different placement or setup when they are on the secondary. 

The way I thought to do this would be to set up the Main Account (Primary) and copy the Blish hud setup somehow into the Alt Account (Primary) profile so everything is matching, but it'd be cool if there is some way to automatically have those sync together while keeping their own APIs assigned.

Another thing would be syncing up the to-do module so they are ticked off across the board, but I don't think that's possible to do, which is why I am asking for the best way to go about doing all of this and if there are any ways to automate it. That's also why I assumed the to-do wouldn't but possible to set up this way and I'd have to stick to either only using the to-do in one profile or using a different program to do that, but if there were any good ways to make it work cleaner and better I'm all ears.

**Comment by Kawsmecal** [comment]
Another way to see it:

Profile A: Main API + Main Screen Setup
Profile B: Alt API + Alt Screen Setup
Profile C: Main API + Alt Screen Setup
Profile D: Alt API + Main Screen Setup

I want to have the setup placements linked on A&D and B&C while keeping the APIs matching on profiles A&C and B&D, but I think the only way to do this is to set them up on each screen and copy over settings and change the API. I'm not sure exactly where to do this to copy over everything or if this is the most optimized way to do it. IDK if I can have the Mumble link setup for the same thing with both APIs put in and it automatically knows the API that matches the account of the client it's over?

**Comment by Healix** [comment]
Settings > Guild Wars 2 > Launch options
 - Enable the mumble link name with the default value (MumbleLink_%accountid%)
 - At the bottom is where you can configure programs to run after launching for all accounts. If you have Blish added here, remove it

Select all of the accounts (A, B, C, D). You can hold CTRL and click to select accounts, or CTRL+A to select them all. With multiple accounts selected, you can right click a selected account > Edit (multiple). Under the launch options, add Blish to run after launching with the option:

-m MumbleLink_%accountid%

Now select only B and D, right click > Edit (multiple). Under the launch options, edit Blish. Change the option to:

-m MumbleLink_%accountid% -s "%userprofile%\Documents\Guild Wars 2\addons\blishhud2"

Open the folder where Blish saves its settings (%userprofile%\Documents\Guild Wars 2\addons) and make a copy of the blishhud folder, naming it blishhud2

A and C will now share their settings and use the default folder, while B and D will share their settings using the blishhud2 folder. If you add a module to A, C will have it, but B and D won't.

**Comment by Kawsmecal** [comment]
> Settings > Guild Wars 2 > Launch options
> 
> * Enable the mumble link name with the default value (MumbleLink_%accountid%)
> * At the bottom is where you can configure programs to run after launching for all accounts. If you have Blish added here, remove it
> 
> Select all of the accounts (A, B, C, D). You can hold CTRL and click to select accounts, or CTRL+A to select them all. With multiple accounts selected, you can right click a selected account > Edit (multiple). Under the launch options, add Blish to run after launching with the option:
> 
> -m MumbleLink_%accountid%
> 
> Now select only B and D, right click > Edit (multiple). Under the launch options, edit Blish. Change the option to:
> 
> -m MumbleLink_%accountid% -s "%userprofile%\Documents\Guild Wars 2\addons\blishhud2"
> 
> Open the folder where Blish saves its settings (%userprofile%\Documents\Guild Wars 2\addons) and make a copy of the blishhud folder, naming it blishhud2
> 
> A and C will now share their settings and use the default folder, while B and D will share their settings using the blishhud2 folder. If you add a module to A, C will have it, but B and D won't.

Hell yeah thanks a lot!

**Comment by bluenote73** [comment]
I believe some of the blish addons complain/break if you do it this way. I was forced to have separate everything for instances I wanted blish in, but that was a little while back so I can't say for sure it wasn't fixed or changed. So just FWIW, at some point it may break for you triggered by installing one or other module.

**Comment by Kawsmecal** [comment]
> 

Yeah it broke for me after this so I just started using Blish alone and relogging so it'd swap over to the one I wanted on my main screen. I hope there's a way to make Blish automatically switch to the active game but it doesn't seem to want to behave that way.
